app/models/institutions_archive.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_dependency 'institution'
+
+class InstitutionsArchive < Institution
+  self.table_name = 'institutions_archives'
+
+  # class methods
+  def self.archive(version)
+    version_number = version.number
+
+    conn = ActiveRecord::Base.connection
+
+    begin
+      ActiveRecord::Base.transaction do
+        create_archives(version_number, conn)
+        cleanup_institutions(version_number, conn)
+      end
+    rescue ActiveRecord::StatementInvalid => e
+      notice = 'There was an error occurring at the database level'
+      error_msg = e.original_exception.result.error_message
+      Rails.logger.error "#{notice}: #{error_msg}"
+    rescue StandardError => e
+      notice = 'There was an error of unexpected origin'
+      error_msg = e.message
+      Rails.logger.error "#{notice}: #{error_msg}"
+    end
+
+    conn.close
+  end
+
+  def self.create_archives(version_number, conn)
+    str = <<-SQL
+      INSERT INTO institutions_archives
+        SELECT *
+          FROM institutions
+          WHERE version < ? ;
+    SQL
+
+    sql = InstitutionsArchive.send(:sanitize_sql_for_conditions, [str, version_number])
+    conn.execute(sql)
+  end
+
+  def self.cleanup_institutions(version_number, conn)
+    str = 'DELETE FROM institutions WHERE version < ? ;'
+
+    sql = Institution.send(:sanitize_sql_for_conditions, [str, version_number])
+    conn.execute(sql)
```

Comment:
Let's use `ActiveRecord` where we can.
This whole method definition can be replace with
`Institution.where(version: version_number).delete_all`

Looking at this did prompt me to look at the `Institution` model and I noticed it has this association:
`has_many :yellow_ribbon_programs, dependent: :destroy`
Any thoughts on what we want to do with the associated `yellow_ribbon_programs` records?

Reply:
For the `yellow_ribbon_programs` records, we can leave them in the table